src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -827,46 +832,41 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
     forceFindOpenFileOrRead filePath |> Result.map (fun f -> f.Lines)
 
 
-  let getProjectOptionsForFile' (filePath: string<LocalPath>) file =
-    // TODO Optimize this for better performance
-    // typing #r "nuget: ... " seems to cause GetProjectOptionsFromScript getting called often
-    // which I suspect is also calling dotnet restore too many times
-    // Cancelling doesn't seem to be working as intended here.
-    aval {
-      if Utils.isAScript (UMX.untag filePath) then
-        let! checker = checker
-        and! tfmConfig = tfmConfig
-        and! cts = openFilesTokens |> AMap.tryFindA filePath
+  let openFilesToChangesAndProjectOptions =
```

Comment:
Now we have a map with the key of the filepath and a tuple of the file with changes and it's project options since typically you need both to do anything useful (like type checking)